tests/unit_tests/data_steward/retraction/create_deid_map_test.py

```diff
@@ -0,0 +1,125 @@
+# Python imports
+import unittest
+
+# Third party imports
+import mock
+import pandas as pd
+
+# Project imports
+import retraction.create_deid_map as create_deid_map
+import utils.bq
+
+
+class CreateDeidMapTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        print('**************************************************************')
+        print(cls.__name__)
+        print('**************************************************************')
+
+    def setUp(self):
+        self.project_id = 'test_project_id'
+        self.dataset_id = 'test_dataset_id'
+        self.all_datasets = [
+            'R2021Q1R1_combined', 'R2021Q1R1_deid', 'R2021Q1R2_combined',
+            '2021q1r3_combined', 'R2021Q1R3_deid'
+        ]
+        self.deid_datasets = ['R2021Q1R1_deid', 'R2021Q1R3_deid']
+        self.combined_datasets = [
+            'R2021Q1R1_combined', 'R2021Q1R2_combined', '2021q1r3_combined'
+        ]
+
+    @mock.patch('retraction.create_deid_map.utils.bq.list_datasets')
+    def test_get_combined_datasets_for_deid_map(self, mock_list_datasets):
+        result = create_deid_map.get_combined_datasets_for_deid_map(
+            self.project_id)
+
+        expected_all_datasets = mock_list_datasets.return_value = utils.bq.list_datasets(
+            self.project_id)
+        expected_deid_datasets = self.deid_datasets
+        expected_combined_datasets = create_deid_map.get_corresponding_combined_dataset(
+            expected_all_datasets, expected_deid_datasets)
+
+        self.assertListEqual(expected_combined_datasets, result)
+
+    def test_get_corresponding_combined_dataset(self):
+        result = create_deid_map.get_corresponding_combined_dataset(
+            self.all_datasets, self.deid_datasets)
+        expected = list()
+
+        for d in self.deid_datasets:
+            release = d.split('_')[0]
+            older_release = release[1:]
+            combined, older_combined = release + '_combined', older_release + '_combined'
+
+            if combined in self.all_datasets:
+                expected.append(combined)
+            if older_combined in self.all_datasets:
+                expected.append(older_combined)
+            # Remove duplicates
+        expected = list(set(expected))
```

Comment:
This looks like the same logic as the function itself. This can cause the test to miss possible errors in the function since they will always be in agreement. I would suggest maybe hardcoding the expected combined datasets for now and checking if the function generates what you expect.

Reply:
will make this change